File: 413-arithmetic-slices/413-arithmetic-slices.java
class Solution {
    public int numberOfArithmeticSlices(int[] nums) {
        if (nums.length < 3){return 0;}
        ArrayList<Integer> ans = new ArrayList<>();
        
        for(int i = 0; i < nums.length-2; i++){
            int count = 1;
            int diff = nums[i+1] - nums[i];
            for(int j = i+1 ;j < nums.length; j++){
                if(nums[j] - nums[j-1] == diff){
                    count += 1;
                    if(count >= 3){ans.add(count);}}
                else{break;}
            }
        }
        
        return ans.size();
    }
}

File: 413-arithmetic-slices/413-arithmetic-slices.py
class Solution:
    def numberOfArithmeticSlices(self, nums: List[int]) -> int:
        ans = []
        n = len(nums)
        if len(nums) < 3:
            return 0
        
        for i in range(n-2):
            count = 1
            diff = nums[i+1] - nums[i]
            for j in range(i+1,n):
                if nums[j] - nums[j-1] == diff:
                    count += 1
                    if count >= 3:
                        ans.append(count)
                else:
                    break
                
        return len(ans)
            
            